clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java

```diff
@@ -170,25 +171,27 @@ public HttpClientBuilder withConnectionTimeout(int connectionTimeoutMillis) {
 
   @SuppressWarnings({"SwitchStatementWithTooFewBranches", "unchecked"})
   @Override
-  public <API extends NessieApi> API build(NessieApiVersion apiVersion, Class<API> apiContract) {
+  public <API extends NessieApi> API build(Class<API> apiContract) {
+    Objects.requireNonNull(apiContract, "API contract class must be non-null");
     NessieHttpClient client =
         new NessieHttpClient(
             uri, authentication, tracing, readTimeoutMillis, connectionTimeoutMillis);
-    API api;
+    NessieApiVersion apiVersion;
+    try {
+      apiVersion = (NessieApiVersion) apiContract.getField("NESSIE_API_VERSION").get(null);
```

Comment:
I wonder whether we actually really need the enum itself. Maybe it's sufficient to just do an `NessieApiV1.class.isAssignableFrom(apiContract)` check?

I imagine us later having `NessieApiV1_1.class` / `NessieApiV2.class` and so on and this seems like it would be sufficient?

Reply:
b10146a2c50e1b60e4a85971fa671163d5927734 removes the enum entirely